tests/Unit/PointwiseFunctions/Xcts/Test_AdmMass.cpp

```diff
@@ -45,44 +29,48 @@
 
 namespace {
 
+using Schwarzschild = Xcts::Solutions::Schwarzschild;
 using KerrSchild = Xcts::Solutions::WrappedGr<gr::Solutions::KerrSchild>;
 
-void test_mass_surface_integral(const double distance,
-                                const size_t refinement_level,
-                                const size_t polynomial_order) {
-  // Define black hole parameters
-  const double mass = 1;
-  const double boost_speed = 0.5;
-  const double lorentz_factor = 1. / sqrt(1. - square(boost_speed));
-
-  // Get Kerr-Schild solution
-  const std::array<double, 3> dimensionless_spin{{0., 0., 0.}};
-  const std::array<double, 3> center{{0., 0., 0.}};
-  const std::array<double, 3> boost_velocity{{0., 0., boost_speed}};
-  const KerrSchild solution{mass, dimensionless_spin, center, boost_velocity};
+template <typename... RequestedTags>
+tuples::TaggedTuple<RequestedTags...> get_solution_variables(
+    const std::variant<Schwarzschild, KerrSchild>& solution,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& coords,
+    const tmpl::list<RequestedTags...> tags) {
+  if (auto* schwarzschild = std::get_if<Schwarzschild>(&solution)) {
+    return schwarzschild->variables(coords, tags);
+  }
+  if (auto* kerrschild = std::get_if<KerrSchild>(&solution)) {
+    return kerrschild->variables(coords, tags);
+  }
+  return tuples::TaggedTuple<RequestedTags...>();
+}
 
+void test_mass_surface_integral(
+    const double distance, const double mass, const double boost_speed,
+    const std::variant<Schwarzschild, KerrSchild> solution) {
   // Set up domain
+  const size_t h_refinement = 1;
+  const size_t p_refinement = 6;
   domain::creators::Sphere shell{
-      /* inner_radius */ 2 * mass,
+      /* inner_radius */ 0.5 * mass,
```

Comment:
You can take the horizon radius or inner radius as an argument to this function (though currently it does not matter because you don't do a volume integral)